### PR TITLE
Database

### PR DESCRIPTION
**Pull request by dxlpi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,11 @@
 cmake_minimum_required(VERSION 3.20)
+# suppress warning
+cmake_policy(SET CMP0167 NEW)
```
Comment (rgnter) [inline CMakeLists.txt:3]: Can't this just be a command line parameter specific for your version for now?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```
Comment (rgnter) [inline CMakeLists.txt:7]: Redundant, set by `target_compile_features`.

File: CMakeLists.txt [hunk 3]
```diff
@@ -59,4 +66,21 @@ endif ()
...
+
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBPQXX REQUIRED libpqxx)
+
+include_directories(${LIBPQXX_INCLUDE_DIRS})
+link_directories(${LIBPQXX_LIBRARY_DIRS})
+
+# Make libpqxx available globally
+set(PROJECT_LIBRARIES ${PROJECT_LIBRARIES} ${LIBPQXX_LIBRARIES})
+
```
Comment (rgnter) [inline CMakeLists.txt:79]: Don't use `PkgConfig`, have the library cloned to 3rd-party folder as a submodule.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -15,4 +15,14 @@ target_link_libraries(test_buffers
...
+pkg_check_modules(LIBPQXX REQUIRED libpqxx)
 
+include_directories(${LIBPQXX_INCLUDE_DIRS})
+link_directories(${LIBPQXX_LIBRARY_DIRS})
+
+add_executable(test_executable src/TestMagic.cpp)
+target_link_libraries(test_executable ${LIBPQXX_LIBRARIES})
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (rgnter) [inline tests/CMakeLists.txt:28]: Redundant

File: tests/test.cpp [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+#include <thread>
+
+int main() {
+    std::jthread t([] { std::cout << "Hello, jthread!" << std::endl; });
+    return 0;
+}
```
Comment (rgnter) [inline tests/test.cpp:7]: Not needed, right?

**Review by rgnter — CHANGES_REQUESTED** [review]

File: tests/test.cpp (hunk 5 above)
Comment (dxlpi) [reply tests/test.cpp:7]: Made all the changes!

File: CMakeLists.txt (hunk 1 above)
Comment (rgnter) [reply CMakeLists.txt]: Thank you

File: CMakeLists.txt (hunk 3 above)
Comment (rgnter) [reply CMakeLists.txt]: Thank you

File: tests/CMakeLists.txt (hunk 4 above)
Comment (rgnter) [reply tests/CMakeLists.txt]: Thank you

File: tests/test.cpp (hunk 5 above)
Comment (rgnter) [reply tests/test.cpp:7]: Thank you

File: 3rd-party/CMakeLists.txt [hunk 6]
```diff
@@ -1 +1 @@
-add_subdirectory(spdlog)
\ No newline at end of file
+add_subdirectory(spdlog)
```
Comment (rgnter) [inline 3rd-party/CMakeLists.txt:1]: Add to this file:
```cmake
add_subdirectory(libpqxx)
```

And in the base cmakelists.txt file, link alicia-libserver against the `libpqxx::libpqxx` target

File: tests/test.cpp (hunk 5 above)
Comment (rgnter) [reply tests/test.cpp:7]: Still outstanding, the file is still present 😔

**Review by rgnter — CHANGES_REQUESTED** [review]

**Review by rgnter — APPROVED** [review]

**Pull request merged**